python/lsst/pex/config/callStack.py

```diff
@@ -0,0 +1,158 @@
+#
+# LSST Data Management System
+# Copyright 2017 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import print_function, division, absolute_import
+
+import re
+import os
+import inspect
+import linecache
+
+__all__ = ['getStackFrame', 'StackFrame', 'getCallStack']
+
+
+def getStackFrame(relative=0):
+    """Retrieve the stack frame for the caller
+
+    By "caller", we mean our user's caller.
+
+    Parameters
+    ----------
+    relative : `int`, non-negative
+        Number of frames above the caller to retrieve.
+    """
+    frame = inspect.currentframe().f_back.f_back  # Our caller's caller
+    for ii in range(relative):
+        frame = frame.f_back
```

Comment:
Sorry, neglected to address this before merging.
I deliberately didn't add the `None` check, so as to keep the code simple for optimal performance (this gets called tens of millions of times in a ci_hsc run), and because it's not necessary: if we go too far up the stack, an exception will automatically result as we attempt to access an attribute on `None`, and it should then be very clear what the problem is.